doc/src/guides/solving/solve-system-of-equations-algebraically.md

```diff
@@ -0,0 +1,130 @@
+# Solve a System of Equations Algebraically
+
+Use SymPy to solve a system of equations algebraically. For example, $x^2 + y =
+2z, y = -4z$ yields $\{(x = -\sqrt{6z}, y = -4z),$ ${(x = \sqrt{6z}, y =
+-4z)\}}$.
+
+Alternatives to consider:
+- Some systems of equations cannot be solved algebraically (either at all or by
+SymPy), so you may have to {func}`solve your system of equations numerically
+using nsolve() <sympy.solvers.solvers.nsolve>` instead.
+
+Here is an example of solving a system of equations algebraically:
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> solve([x**2 + y - 2*z, y + 4*z], x, y, dict=True)
+[{x: -sqrt(6)*sqrt(z), y: -4*z}, {x: sqrt(6)*sqrt(z), y: -4*z}]
+```
+
+## Guidance
+
+Refer to
+[](solving-guidance.md#include-the-variable-to-be-solved-for-in-the-function-call)
+and [](solving-guidance.md#ensure-consistent-formatting-from).
+
+There are two methods below for containing solution results: dictionary or set.
+A dictionary is easier to interrogate programmatically, so if you need to
+extract solutions using code, we recommend the dictionary approach.
+
+## Solve and Use Results in a Dictionary
+
+### Solve Into a Dictionary of Solutions
+
+You can solve a system of equations for some variables (for example, $x$ and
+$y$) leaving another symbol as a variable (for example, $z$). You can specify
+the variables to solve for as multiple separate arguments, or as a list (or
+tuple):
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> equations = [x**2 + y - 2*z, y + 4*z]
+>>> solutions = solve(equations, x, y, dict=True)
+>>> solutions
+[{x: -sqrt(6)*sqrt(z), y: -4*z}, {x: sqrt(6)*sqrt(z), y: -4*z}]
+```
+
+### Use a Dictionary of Solutions
+
+You can then extract solutions by specifying in brackets ("slicing") the
+solution number, and then the symbol. For example `solutions[0][x]` gives the
+result for `x` in the first solution:
+
+```py
+>>> solutions[0][x]
+-sqrt(6)*sqrt(z)
+>>> solutions[0][y]
+-4*z
+```
+
+## Solve Results in a Set
+
+To get a list of symbols and set of solutions, use `set=True` instead of
+`dict=True`:
+
+```py
+from sympy import solve
+from sympy.abc import x, y, z
+solve([x**2 + y - 2*z, y + 4*z], [x, y], set=True)
+([x, y], {(-sqrt(6)*sqrt(z), -4*z), (sqrt(6)*sqrt(z), -4*z)})
+```
+
+## Options That Can Speed up {func}`~.solve`
+
+Refer to [](solving-guidance.md#options-that-can-speed-up).
+
+## Not All Systems of Equations Can be Solved
+
+### Systems of Equations With no Solution
+
+Some systems of equations have no solution. For example, the following system
+reduces to $z = 2z$, which has no general solution:
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y, z
+>>> solve([x + y - z, x + y - 2*z], [x, y], dict=True)
+[]
+```
```

Comment:
This is probably a confusing example because whether or not it has solutions depends on the value of `z` i.e. if `z=0` then there are solutions. It would be better to have something that reduces to `z = z + 1` which definitely doesn't have solutions for any possible value of z.

I do think it would be good somewhere else on this page though to explain how solve etc handle the case where solutions exist conditionally dependent on the (unknown) values of any symbolic parameters:
https://stackoverflow.com/questions/73611001/sympy-linsolve-with-empty-output-for-a-system-of-overdetermined-equations
The example you have above is a good one for explaining that `solve` will not assume that `z=0` even if that's the only value of `z` that makes the system of equations consistent because `z` is a parameter rather than an unkown.